server/exercise/src/main/scala/com/eigengo/lift/exercise/MultiPacketDecoder.scala

```diff
@@ -0,0 +1,74 @@
+package com.eigengo.lift.exercise
+
+import java.nio.ByteBuffer
+
+import scodec.bits.{BitVector, ByteOrdering, ByteVector}
+
+import scalaz.\/
+
+/**
+ * The multi packet message follows a simple structure
+ *
+ * {{{
+ *    header: UInt16     = 0xcab0 // 2
+ *    count: Byte        = ...    // 3
+ *    ===
+ *    size0: UInt16      = ...    // 5
+ *    sloc0: Byte        = ...    // 6
+ *    data0: Array[Byte] = ...
+ *    size1: UInt16      = ...
+ *    sloc1: Byte        = ...
+ *    data1: Array[Byte] = ...
+ *    ...
+ *    sizen: UInt16      = ...
+ *    slocn: Byte        = ...
+ *    datan: Array[Byte] = ...
+ * }}}
+ */
+object MultiPacketDecoder {
+  private val header = 0xcab0.toShort
+
+  def decodeShort(b0: Byte, b1: Byte): Int = {
+    ByteVector(b0, b1).toInt(signed = false, ordering = ByteOrdering.BigEndian)
+  }
+
+  def decodeSensorDataSourceLocation(sloc: Byte): String \/ SensorDataSourceLocation = sloc match {
+    case 0x01 ⇒ \/.right(SensorDataSourceLocationWrist)
```

Comment:
Scalaz offers two more syntexes to do the same (if preferred). 
`\/-(SensorDataSourceLocationWrist)`
and
`SensorDataSourceLocationWrist.right`